wallet/wallet.c

```diff
@@ -4706,3 +4706,92 @@ void wallet_offer_mark_used(struct db *db, const struct sha256 *offer_id)
 		offer_status_update(db, offer_id, status, newstatus);
 	}
 }
+
+bool wallet_datastore_add(struct wallet *w, const char *key, const u8 *data)
+{
+	struct db_stmt *stmt;
+
+	/* Test if already exists. */
+	stmt = db_prepare_v2(w->db, SQL("SELECT 1"
+					"  FROM datastore"
+					" WHERE key = ?;"));
+	db_bind_text(stmt, 0, key);
+	db_query_prepared(stmt);
+
+	if (db_step(stmt)) {
+		tal_free(stmt);
+		return false;
+	}
```

Comment:
I don't know if I'm missing some things in the design of this functionality, but if a plugin needs to override functionality is it now possible with the current API of the wallet.h, right?

Do you think that is necessary use some additional parameters to the RPC command to give the possibility to override the propriety with key?

Somethings like `lightning-cli datastore "fooo" "some" true`

Reply:
Yes, since this isn't going to be in this release, I can revise this properly.

I am going to allow both hex and string arguments, and also allow a mode argument:
- "must-create", "must-replace", or "create-or-replace", "must-append", "create-or-append".

Reply:
I like this idea Rusty.